src/main/java/org/spongepowered/common/registry/type/entity/CareerRegistryModule.java

```diff
@@ -61,28 +49,20 @@ public boolean allowsApiRegistration() {
 
     @Override
     public void registerAdditionalCatalog(Career extraCatalog) {
-        if (this.forgeSpongeMapping.containsKey(extraCatalog.getId().toLowerCase())) {
-            // Basically, forge has alternate names for a minor few vanilla
-            // careers and this avoids having duplicate "careers" registered.
-            return;
-        }
-        if (!this.careerMap.containsKey(extraCatalog.getId())) {
-            this.careerMap.put(extraCatalog.getId().toLowerCase(), extraCatalog);
-        }
-        ProfessionRegistryModule.getInstance().registerCareerForProfession(extraCatalog);
+        tryRegister(extraCatalog);
```

Comment:
I tested it with 
* forge-1.8.9-11.15.1.1722-universal.jar
* Sponge 1140

Before fix:
````log
[DEBUG]: osc.entity.SpongeCareer{id=cleric, name=Cleric, translation=SpongeTranslation{id=entity.Villager.cleric}, toString=SpongeCareer{id=cleric, name=Cleric, type=0, translation=SpongeTranslation{id=entity.Villager.cleric}, profession=SpongeProfession{id=priest, name=priest, type=2}}}
[ERROR]: Failed to obtain instance for Career with id cleric
[DEBUG]: osc.entity.SpongeCareer{id=fletcher, name=Fletcher, translation=SpongeTranslation{id=entity.Villager.fletcher}, toString=SpongeCareer{id=fletcher, name=Fletcher, type=3, translation=SpongeTranslation{id=entity.Villager.fletcher}, profession=SpongeProfession{id=farmer, name=farmer, type=0}}}
[ERROR]: Failed to obtain instance for Career with id fletcher
[DEBUG]: osc.entity.SpongeCareer{id=armorer, name=Armorer, translation=SpongeTranslation{id=entity.Villager.armor}, toString=SpongeCareer{id=armorer, name=Armorer, type=0, translation=SpongeTranslation{id=entity.Villager.armor}, profession=SpongeProfession{id=blacksmith, name=blacksmith, type=3}}}
[ERROR]: Failed to obtain instance for Career with id armorer
[DEBUG]: osc.entity.SpongeCareer{id=weapon_smith, name=Weapon Smith, translation=SpongeTranslation{id=entity.Villager.weapon}, toString=SpongeCareer{id=weapon_smith, name=Weapon Smith, type=1, translation=SpongeTranslation{id=entity.Villager.weapon}, profession=SpongeProfession{id=blacksmith, name=blacksmith, type=3}}}
[ERROR]: Failed to obtain instance for Career with id weapon_smith
[DEBUG]: osc.entity.SpongeCareer{id=tool_smith, name=Tool Smith, translation=SpongeTranslation{id=entity.Villager.tool}, toString=SpongeCareer{id=tool_smith, name=Tool Smith, type=2, translation=SpongeTranslation{id=entity.Villager.tool}, profession=SpongeProfession{id=blacksmith, name=blacksmith, type=3}}}
[ERROR]: Failed to obtain instance for Career with id tool_smith
[DEBUG]: osc.entity.SpongeCareer{id=farmer, name=Farmer, translation=SpongeTranslation{id=entity.Villager.farmer}, toString=SpongeCareer{id=farmer, name=Farmer, type=0, translation=SpongeTranslation{id=entity.Villager.farmer}, profession=SpongeProfession{id=farmer, name=farmer, type=0}}}
[ERROR]: Failed to obtain instance for Career with id farmer
[DEBUG]: osc.entity.SpongeCareer{id=librarian, name=Librarian, translation=SpongeTranslation{id=entity.Villager.librarian}, toString=SpongeCareer{id=librarian, name=Librarian, type=0, translation=SpongeTranslation{id=entity.Villager.librarian}, profession=SpongeProfession{id=librarian, name=librarian, type=1}}}
[ERROR]: Failed to obtain instance for Career with id librarian
[DEBUG]: osc.entity.SpongeCareer{id=butcher, name=Butcher, translation=SpongeTranslation{id=entity.Villager.butcher}, toString=SpongeCareer{id=butcher, name=Butcher, type=0, translation=SpongeTranslation{id=entity.Villager.butcher}, profession=SpongeProfession{id=butcher, name=butcher, type=4}}}
[ERROR]: Failed to obtain instance for Career with id butcher
[DEBUG]: osc.entity.SpongeCareer{id=shepherd, name=Shepherd, translation=SpongeTranslation{id=entity.Villager.shepherd}, toString=SpongeCareer{id=shepherd, name=Shepherd, type=2, translation=SpongeTranslation{id=entity.Villager.shepherd}, profession=SpongeProfession{id=farmer, name=farmer, type=0}}}
[ERROR]: Failed to obtain instance for Career with id shepherd
[DEBUG]: osc.entity.SpongeCareer{id=fisherman, name=Fisherman, translation=SpongeTranslation{id=entity.Villager.fisherman}, toString=SpongeCareer{id=fisherman, name=Fisherman, type=1, translation=SpongeTranslation{id=entity.Villager.fisherman}, profession=SpongeProfession{id=farmer, name=farmer, type=0}}}
[ERROR]: Failed to obtain instance for Career with id fisherman
[DEBUG]: osc.entity.SpongeCareer{id=leatherworker, name=Leatherworker, translation=SpongeTranslation{id=entity.Villager.leather}, toString=SpongeCareer{id=leatherworker, name=Leatherworker, type=1, translation=SpongeTranslation{id=entity.Villager.leather}, profession=SpongeProfession{id=butcher, name=butcher, type=4}}}
[ERROR]: Failed to obtain instance for Career with id leatherworker
````

After fix:
````log
[DEBUG]: osc.entity.SpongeCareer{id=minecraft:fletcher, name=Fletcher, translation=SpongeTranslation{id=entity.Villager.fletcher}, toString=SpongeCareer{id=minecraft:fletcher, name=Fletcher, translation=SpongeTranslation{id=entity.Villager.fletcher}, type=3, profession=SpongeProfession{id=farmer, name=farmer, type=0}}}
[DEBUG]: osc.entity.SpongeCareer{id=minecraft:butcher, name=Butcher, translation=SpongeTranslation{id=entity.Villager.butcher}, toString=SpongeCareer{id=minecraft:butcher, name=Butcher, translation=SpongeTranslation{id=entity.Villager.butcher}, type=0, profession=SpongeProfession{id=butcher, name=butcher, type=4}}}
[DEBUG]: osc.entity.SpongeCareer{id=minecraft:tool_smith, name=Tool Smith, translation=SpongeTranslation{id=entity.Villager.tool}, toString=SpongeCareer{id=minecraft:tool_smith, name=Tool Smith, translation=SpongeTranslation{id=entity.Villager.tool}, type=2, profession=SpongeProfession{id=blacksmith, name=blacksmith, type=3}}}
[DEBUG]: osc.entity.SpongeCareer{id=minecraft:shepherd, name=Shepherd, translation=SpongeTranslation{id=entity.Villager.shepherd}, toString=SpongeCareer{id=minecraft:shepherd, name=Shepherd, translation=SpongeTranslation{id=entity.Villager.shepherd}, type=2, profession=SpongeProfession{id=farmer, name=farmer, type=0}}}
[DEBUG]: osc.entity.SpongeCareer{id=minecraft:weapon_smith, name=Weapon Smith, translation=SpongeTranslation{id=entity.Villager.weapon}, toString=SpongeCareer{id=minecraft:weapon_smith, name=Weapon Smith, translation=SpongeTranslation{id=entity.Villager.weapon}, type=1, profession=SpongeProfession{id=blacksmith, name=blacksmith, type=3}}}
[DEBUG]: osc.entity.SpongeCareer{id=minecraft:armorer, name=Armorer, translation=SpongeTranslation{id=entity.Villager.armor}, toString=SpongeCareer{id=minecraft:armorer, name=Armorer, translation=SpongeTranslation{id=entity.Villager.armor}, type=0, profession=SpongeProfession{id=blacksmith, name=blacksmith, type=3}}}
[DEBUG]: osc.entity.SpongeCareer{id=minecraft:farmer, name=Farmer, translation=SpongeTranslation{id=entity.Villager.farmer}, toString=SpongeCareer{id=minecraft:farmer, name=Farmer, translation=SpongeTranslation{id=entity.Villager.farmer}, type=0, profession=SpongeProfession{id=farmer, name=farmer, type=0}}}
[DEBUG]: osc.entity.SpongeCareer{id=minecraft:cleric, name=Cleric, translation=SpongeTranslation{id=entity.Villager.cleric}, toString=SpongeCareer{id=minecraft:cleric, name=Cleric, translation=SpongeTranslation{id=entity.Villager.cleric}, type=0, profession=SpongeProfession{id=priest, name=priest, type=2}}}
[DEBUG]: osc.entity.SpongeCareer{id=minecraft:librarian, name=Librarian, translation=SpongeTranslation{id=entity.Villager.librarian}, toString=SpongeCareer{id=minecraft:librarian, name=Librarian, translation=SpongeTranslation{id=entity.Villager.librarian}, type=0, profession=SpongeProfession{id=librarian, name=librarian, type=1}}}
[DEBUG]: osc.entity.SpongeCareer{id=minecraft:fisherman, name=Fisherman, translation=SpongeTranslation{id=entity.Villager.fisherman}, toString=SpongeCareer{id=minecraft:fisherman, name=Fisherman, translation=SpongeTranslation{id=entity.Villager.fisherman}, type=1, profession=SpongeProfession{id=farmer, name=farmer, type=0}}}
[DEBUG]: osc.entity.SpongeCareer{id=minecraft:leatherworker, name=Leatherworker, translation=SpongeTranslation{id=entity.Villager.leather}, toString=SpongeCareer{id=minecraft:leatherworker, name=Leatherworker, translation=SpongeTranslation{id=entity.Villager.leather}, type=1, profession=SpongeProfession{id=butcher, name=butcher, type=4}}}
````

After some debugging I noticed, that `registerAdditionalCatalog(Career extraCatalog)` and `registerCareer(Career career)` are not called at all. So my question is: Is Forge's MixinVillagerProfession working at all?